### PR TITLE
Allow generating archive with multiple resource types

### PR DESCRIPTION
**Pull request by const-cloudinary**

### CODE REVIEW

File: Shared/Actions/ArchiveParams.cs [hunk 1]
```diff
@@ -48,6 +49,23 @@ public ArchiveParams PublicIds(List<string> publicIds)
             return this;
         }
 
+        /// <summary>
+        /// Get a list of Fully Qualified Public IDs for the specific assets to be included in the archive
+        /// </summary>
+        public List<string> FullyQualifiedPublicIds()
+        {
+            return m_publicIds;
```
Comment (asisayag2) [inline Shared/Actions/ArchiveParams.cs]: @const-cloudinary , replace with m_fullyQualifiedPublicIds

File: Shared/Actions/ArchiveParams.cs [hunk 2]
```diff
@@ -89,13 +107,14 @@ public ArchiveParams Prefixes(List<string> prefixes)
         public override void Check()
         {
             if ((m_publicIds == null || m_publicIds.Count == 0) &&
+                (m_fullyQualifiedPublicIds == null || m_fullyQualifiedPublicIds.Count == 0) &&
```
Comment (asisayag2) [inline Shared/Actions/ArchiveParams.cs]: @const-cloudinary , please add a validation for auto && fully qualified names. 
Also check the mutually exclusive behavior

File: Shared/Actions/ArchiveParams.cs [hunk 3]
```diff
@@ -83,19 +101,33 @@ public ArchiveParams Prefixes(List<string> prefixes)
...
-                (m_tags == null || m_tags.Count == 0))
-                throw new ArgumentException("At least one of the following \"filtering\" parameters needs to be specified: PublicIds, Tags or Prefixes.");
+            if (m_publicIds?.Any() != true &&
+                m_fullyQualifiedPublicIds?.Any() != true &&
+                m_prefixes?.Any() != true &&
+                m_tags?.Any() != true)
+                throw new ArgumentException("At least one of the following \"filtering\" parameters needs " +
+                                            "to be specified: PublicIds, FullyQualifiedPublicIds, Tags or Prefixes.");
+
+            if (m_resourceType == "auto")
```
Comment (asisayag2) [inline Shared/Actions/ArchiveParams.cs]: @const-cloudinary maybe:
if (!(m_resourceType=="auto" ^  m_fullyQualifiedPublicIds?.Any())) throw new ArgumentException ("Archiving assets from different resource types requires both fully qualified names and setting ResourceType to \"auto\" ")

File: Shared/Actions/ArchiveParams.cs [hunk 4]
```diff
@@ -83,19 +101,33 @@ public ArchiveParams Prefixes(List<string> prefixes)
...
         {
-            if ((m_publicIds == null || m_publicIds.Count == 0) &&
-                (m_prefixes == null || m_prefixes.Count == 0) &&
-                (m_tags == null || m_tags.Count == 0))
-                throw new ArgumentException("At least one of the following \"filtering\" parameters needs to be specified: PublicIds, Tags or Prefixes.");
+            if (m_publicIds?.Any() != true &&
+                m_fullyQualifiedPublicIds?.Any() != true &&
+                m_prefixes?.Any() != true &&
+                m_tags?.Any() != true)
+                throw new ArgumentException("At least one of the following \"filtering\" parameters needs " +
```
Comment (jackieros) [inline Shared/Actions/ArchiveParams.cs]:
```suggestion
                throw new ArgumentException("At least one of the following \'filtering\' parameters must " +
```

File: Shared/Actions/ArchiveParams.cs [hunk 5]
```diff
@@ -83,19 +101,33 @@ public ArchiveParams Prefixes(List<string> prefixes)
...
+                                            "to be specified: PublicIds, FullyQualifiedPublicIds, Tags or Prefixes.");
```
Comment (jackieros) [inline Shared/Actions/ArchiveParams.cs]:
```suggestion
                                            "be specified: PublicIds | FullyQualifiedPublicIds | Tags | Prefixes.");
```

File: Shared/Actions/ArchiveParams.cs (hunk 3 above)
Comment (jackieros) [reply Shared/Actions/ArchiveParams.cs]: To create an archive with multiple types of assets, you must set ResourceType to 'auto' and provide fully-qualified public IDs (For example, 'video/upload/myvideo').

**Pull request merged**